torch/testing/_internal/common_methods_invocations.py

```diff
@@ -15317,6 +15317,8 @@ def reference_flatten(input, start_dim=0, end_dim=-1):
             # lambda impl
             DecorateInfo(unittest.expectedFailure, 'TestJit', 'test_variant_consistency_jit'),
             DecorateInfo(unittest.expectedFailure, 'TestNormalizeOperators', 'test_normalize_operator_exhaustive'),
+            # FIXME: this fails on CI and passes on quansight's qgpu machine
+            DecorateInfo(unittest.expectedFailure, 'TestCommon', 'test_compare_cpu'),
```

Comment:
@amjames The tests are passing after this `expectedFailure` which fails on qgpu with 

```
================================================== 434 passed, 501 skipped, 47841 deselected, 26 xfailed, 11 warnings in 34.78s ==================================================
(pytorch2) srossross@qgpu1:~/pytorch$ py.test test/test_ops.py -v -k test_compare_cpu_nn_functional_embedding_cuda_float32
============================================================================== test session starts ===============================================================================
platform linux -- Python 3.8.13, pytest-7.1.2, pluggy-1.0.0 -- /home/srossross/.conda/envs/pytorch2/bin/python
cachedir: .pytest_cache
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/home/srossross/pytorch/.hypothesis/examples')
rootdir: /home/srossross/pytorch, configfile: pytest.ini
plugins: hypothesis-6.54.4
collected 48802 items / 48801 deselected / 1 selected                                                                                                                            

test/test_ops.py::TestCommonCUDA::test_compare_cpu_nn_functional_embedding_cuda_float32 FAILED                                                                             [100%]

==================================================================================== FAILURES ====================================================================================
______________________________________________________ TestCommonCUDA.test_compare_cpu_nn_functional_embedding_cuda_float32 ______________________________________________________
Unexpected success
============================================================================ short test summary info =============================================================================
FAILED test/test_ops.py::TestCommonCUDA::test_compare_cpu_nn_functional_embedding_cuda_float32
=============================================================== 1 failed, 48801 deselected, 11 warnings in 15.09s ================================================================
```

Reply:
cc @lezcano

Reply:
[here is](https://github.com/pytorch/pytorch/actions/runs/3071172014/jobs/4986626966) the original error 
```
======================================================================
FAIL [0.009s]: test_compare_cpu_nn_functional_embedding_cuda_float32 (__main__.TestCommonCUDA)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/conda/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 1950, in wrapper
    method(*args, **kwargs)
  File "/opt/conda/lib/python3.10/site-packages/torch/testing/_internal/common_device_type.py", line 378, in instantiated_test
    result = test(self, **param_kwargs)
  File "/opt/conda/lib/python3.10/site-packages/torch/testing/_internal/common_device_type.py", line 815, in test_wrapper
    return test(*args, **kwargs)
  File "/opt/conda/lib/python3.10/site-packages/torch/testing/_internal/common_device_type.py", line 975, in only_fn
    return fn(slf, *args, **kwargs)
  File "/opt/conda/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 1283, in wrapper
    fn(*args, **kwargs)
  File "/var/lib/jenkins/workspace/test/test_ops.py", line 187, in test_compare_cpu
    self.assertEqual(cuda_results, cpu_results)
  File "/opt/conda/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 2412, in assertEqual
    assert_equal(
  File "/opt/conda/lib/python3.10/site-packages/torch/testing/_comparison.py", line 1093, in assert_equal
    raise error_metas[0].to_error(msg)
AssertionError: Tensor-likes are not close!

Mismatched elements: 15 / 45 (33.3%)
Greatest absolute difference: 0.5051435921341181 at index (0, 0, 3) (up to 1e-05 allowed)
Greatest relative difference: 0.9674892692698572 at index (0, 0, 2) (up to 1.3e-06 allowed)
```

Reply:
I'd skip it if it's not clear where it fails and I'd open an issue and point the skip to the issue.

Could it be the case that this function is not deterministic for some inputs?

Reply:
added [issue ](https://github.com/pytorch/pytorch/issues/85377)